plone/app/widgets/utils.py

```diff
@@ -51,6 +52,14 @@ class NotImplemented(Exception):
 def get_date_options(request):
     calendar = request.locale.dates.calendars['gregorian']
     today = datetime.today()
+    try:
+        past_years = int(api.portal.get_registry_record(
+            'plone.app.widgets.date_options.past_years'))
+        future_years = int(api.portal.get_registry_record(
+            'plone.app.widgets.date_options.future_years'))
+    except:
```

Comment:
Failing for a specific error is better. We should use ``InvalidParameterError`` from ``plone.api.exc`` instead of non-specific exception clause.